core/src/main/java/org/mule/runtime/core/internal/util/MessagingExceptionResolver.java

```diff
@@ -103,6 +104,31 @@ public MessagingException resolve(final MessagingException me, MuleContext conte
     return enrich(result, failingComponent, event, context);
   }
 
+
+  public ModuleOperationMuleException resolveSmartConnector(final MessagingException me, MuleContext context) {
```

Comment:
rename `me` to `innerCause` or `smartConnectorCause`, something that indicates that the context of that exception is the smart connector